p2p/src/sync/peer/block_manager.rs

```diff
@@ -606,9 +610,6 @@ where
             return Err(P2pError::ProtocolError(ProtocolError::DisconnectedHeaders));
         }
 
-        let last_header = headers.last().expect("Headers shouldn't be empty");
-        self.wait_for_clock_diff(last_header.timestamp()).await;
```

Comment:
Why removing this? Please be careful with such radical changes without studying the consequences.

Reply:
I moved it to the line 644 where the block height is available